File: practice9_3/practice9_3/Rectangle.cpp
#include "Rectangle.h"

Rectangle::Rectangle(double newHeight, double newWidth) {
	height = newHeight;
	width = newWidth;
}
double Rectangle::getArea() {
	return width * height;
	}
double Rectangle::getPerimeter() {
	return (width + height) * 2;
	}
double Rectangle::getWidth() {
	return width;
}
double Rectangle::getHeight() {
	return height;
}
void Rectangle::setWidth(double newWidth) {
	if (width >= 0)
		width = newWidth;
	else
		width = 0;
	}
void Rectangle::setHeight(double newHeight) {
	if (height >= 0)
		height = newHeight;
	else
		height = 0;
	}

File: GeometricObject_Inheritance/Rectangle.py
from GeometricObject import GeometricObject

class Rectangle(GeometricObject):
    def __init__(self, width, height, color, filled):
        self.__width = width
        self.__height = height
        self.setColor(color)
        self.setFilled(filled)

    def getWidth(self):
        return self.__width

    def setWidth(self, width):
        self.__width = width

    def getHeight(self):
        return self.__height

    def setHeight(self, height):
        self.__height = height

    def getArea(self):
        return self.getWidth() * self.getHeight()

    def getPerimeter(self):
        return (self.getWidth() + self.getHeight()) * 2


